### PR TITLE
added function and script to add part comments to part_info

### PR DESCRIPTION
**Pull request by david-deboer**
Now that we have failing parts I need to add to the part_info table.

### CODE REVIEW

File: hera_mc/cm_handling.py [hunk 1]
```diff
@@ -251,8 +251,13 @@ def show_parts(self, part_dossier, verbosity='h'):
...
                     tdata.append(ptsout.strip().strip(','))
-                    if (part_dossier[hpnr]['part_info'] is not None):
-                        comment = part_dossier[hpnr]['part_info'].comment
+                    if len(part_dossier[hpnr]['part_info']):
+                        comment = ''
+                        for pi in part_dossier[hpnr]['part_info']:
+                            if not len(comment):
+                                comment = pi.comment
+                            else:
+                                comment = comment + ', ' + pi.comment
```
Comment (pkgw) [inline hera_mc/cm_handling.py]: This can all be simplified to `comment = ', '.join(part_dossier[hpnr]['part_info'])`.

File: scripts/part_info.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    parser = mc.get_mc_argument_parser()
+    parser.add_argument('-p', '--hpn', help="HERA part number", default=None)
+    parser.add_argument('-r', '--rev', help="Revision of part", default=None)
+    parser.add_argument('-c', '--comment', help="Comment on part", default=None)
+    parser.add_argument('-l', '--library_file', help="Library filename", default=None)
+    cm_utils.add_date_time_args(parser)
+    args = parser.parse_args()
+
+    if args.hpn is None or args.rev is None or args.comment is None:
+        args = query_args(args)
```
Comment (pkgw) [inline scripts/part_info.py]: Given how `query_args` is structured, it can just be called unconditionally.

File: scripts/part_info.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    # Pre-process some args
+    at_date = cm_utils._get_astropytime(args.date, args.time)
+
+    db = mc.connect_to_mc_db(args)
+    session = db.sessionmaker()
+
+    # Check for part
+    print("Adding comment for part {}:{}".format(args.hpn, args.rev))
```
Comment (pkgw) [inline scripts/part_info.py]: This should be phrased more generically as "Adding info for part...", no? One might want to use this command to register a `library_file`.

File: hera_mc/cm_handling.py (hunk 1 above)
Comment (david-deboer) [reply hera_mc/cm_handling.py]: Actually, I only want the .comment field from it so I think I have to loop.

File: hera_mc/cm_handling.py [hunk 4]
```diff
@@ -252,12 +252,10 @@ def show_parts(self, part_dossier, verbosity='h'):
...
                     if len(part_dossier[hpnr]['part_info']):
-                        comment = ''
+                        comment_list = []
                         for pi in part_dossier[hpnr]['part_info']:
-                            if not len(comment):
-                                comment = pi.comment
-                            else:
-                                comment = comment + ', ' + pi.comment
+                                comment_list.append(pi.comment)
+                        comment = ', '.join(comment_list)
```
Comment (pkgw) [inline hera_mc/cm_handling.py]: Ah, right. Could be further tightened up with a list comprehension:

```python
comment = ', '.join(pi.comment for pi in part_dossier[hpnr]['part_info'])
```

**Pull request merged**